CorsixTH/Src/th_pathfind.cpp

```diff
@@ -28,166 +28,154 @@ SOFTWARE.
+        if (TryNode(pNode, iFlags, pNode - iWidth, THTD_North)) return true;
+
+    if(iFlags & THMN_CanTravelS)
+        if (TryNode(pNode, iFlags, pNode + iWidth, THTD_South)) return true;
+
+    return false;
+}
 
-    /* No need to check for the node being on the map edge, as the N/E/S/W
-       flags are set as to prevent travelling off the map (as well as to
-       prevent walking through walls). */
-#define Pathing_Neighbours() \
-    uint32_t iPassable = iFlags & THMN_Passable; \
-    if(iFlags & THMN_CanTravelW) \
-    { \
-        TryNode(pNode - 1, 3); \
-    } \
-    if(iFlags & THMN_CanTravelE) \
-    { \
-        TryNode(pNode + 1, 1); \
-    } \
-    if(iFlags & THMN_CanTravelN) \
-    { \
-        TryNode(pNode - iWidth, 0); \
-    } \
-    if(iFlags & THMN_CanTravelS) \
-    { \
-        TryNode(pNode + iWidth, 2); \
-    } \
-
-#define Pathing_TryNode() \
-    if(iNFlags & THMN_Passable || iPassable == 0) \
-    { \
-        if(pNeighbour->prev == pNeighbour) \
-        { \
-            pNeighbour->prev = pNode; \
-            pNeighbour->distance = pNode->distance + 1; \
-            MakeGuess(pNeighbour); \
-            m_ppDirtyList[m_iDirtyCount++] = pNeighbour; \
-            _openHeapPush(pNeighbour); \
-        } \
-        else if((pNode->distance + 1) < pNeighbour->distance) \
-        { \
-            pNeighbour->prev = pNode; \
-            pNeighbour->distance = pNode->distance + 1; \
-            /* guess doesn't change, and already in the dirty list */ \
-            _openHeapPromote(pNeighbour); \
-        } \
+void BasePathing::pathingTryNode(node_t *pNode, uint32_t iNFlags, uint32_t iPassable, node_t *pNeighbour)
+{
+    if(iNFlags & THMN_Passable || iPassable == 0)
+    {
+        if(pNeighbour->prev == pNeighbour)
+        {
+            pNeighbour->prev = pNode;
+            pNeighbour->distance = pNode->distance + 1;
+            pNeighbour->guess = MakeGuess(pNeighbour);
+            m_pPf->m_ppDirtyList[m_pPf->m_iDirtyCount++] = pNeighbour;
+            m_pPf->_openHeapPush(pNeighbour);
+        }
+        else if(pNode->distance + 1 < pNeighbour->distance)
+        {
+            pNeighbour->prev = pNode;
+            pNeighbour->distance = pNode->distance + 1;
+            /* guess doesn't change, and already in the dirty list */
+            m_pPf->_openHeapPromote(pNeighbour);
+        }
     }
+}
+
+int PathFinder::MakeGuess(node_t *pNode)
+{
+    // As diagonal movement is not allowed, the minimum distance between two
+    // points is the sum of the distance in X and the distance in Y.
+    return abs(pNode->x - m_iEndX) + abs(pNode->y - m_iEndY);
+}
 
-#define Pathing_Next() \
-    if(m_iOpenCount == 0) \
-    { \
-        m_pDestination = NULL; \
-        break; \
-    } \
-    else \
-        pNode = _openHeapPop()
+bool PathFinder::TryNode(node_t *pNode, uint32_t iFlags,
+                         node_t *pNeighbour, int direction)
+{
+    uint32_t iPassable = iFlags & THMN_Passable;
```

Comment:
Nevermind, I see that objectVisitor and idlePathFinder are a little more tricky.